src/include/sof/drivers/timer.h

```diff
@@ -62,6 +62,17 @@ int64_t platform_timer_set(struct timer *timer, uint64_t ticks);
 void platform_timer_clear(struct timer *timer);
 uint64_t platform_timer_get(struct timer *timer);
 uint64_t platform_timer_get_atomic(struct timer *timer);
+
+static inline uint64_t platform_safe_get_time(struct timer *timer)
+{
+	/* Default to something small but at least 1.0 microsecond so it
```

Comment:
Would it not be better to make the default version safe ? (and return time 0 to indicate no timer running)

Reply:
Maybe, I don't know. You tell me. So far I've been spending at least 80% of my time trying to fix issues fairly distant from my actual objectives so this time I tried to minimize the changes but maybe I shouldn't have?

There is a variety of platform_timer_get(timer) implementations. Some implementations don't even look at their timer argument: so these platforms would then lose even more timestamps for no real reason.

At least one platform performs an assert(timer), maybe because it was considered valuable not to keep running with broken timers? Tracing is still useful without timestamps but I wouldn't be so sure about other areas: I can easily imagine some threads stuck forever.

Considering such "diversity" I chose not to open that can of worms and to change only the area in this PR. As of now the new "safe" function is meant to be used only in areas with tricky boot dependencies; like tracing.


> (and return time 0 to indicate no timer running)

As explained in the comment, I'm not returning zero but on small value on purpose: because zero looks like something is wrong.

Reply:
This non-zero value is useful to make the difference with `platform_timer_get(timer)` returning zero (for good or bad reasons). When everyone and everything returns zero including buggy code (but not just), then there's no way to tell where it came from.